tsconfig.json

```diff
@@ -21,5 +21,6 @@
     "skipLibCheck": true,
     "target": "ES2022",
     "jsx": "react-jsx"
-  }
+  },
+  "exclude": ["tools", "node_modules", "lib", "engine", "test", "dist", "docs"]
```

Comment:
Questionable, but advised by vscode typescript to add this.
Improves reference discovery time